cats-backend/src/modules/cats/cats/cats.service.ts

```diff
@@ -28,25 +29,26 @@ export class CatsService {
       const matchingMice = await this.mouseModel.findAll({
         attributes: ['catId'],
         where: {
-          name: { [Op.iLike]: mouseName },
+          name: { [Op.iLike]: `%${mouseName}%` },
         },
-        raw: true,
       });
 
-      miceWithCatIds = [...new Set(matchingMice.map((mouse) => mouse.catId))];
+      miceWithCatIds = Array.from(
+        new Set(matchingMice.map((mouse) => mouse.catId)),
+      );
       if (!miceWithCatIds.length) return [];
     }
 
     const catWhereClause: WhereOptions<Cat> = {};
 
     if (name) {
       catWhereClause[Op.or] = [
-        { firstName: { [Op.iLike]: name } },
-        { lastName: { [Op.iLike]: name } },
+        { firstName: { [Op.iLike]: `%${name}%` } },
+        { lastName: { [Op.iLike]: `%${name}%` } },
```

Comment:
```suggestion
        { firstName: { [Op.iLike]: `%:name%` } },
        { lastName: { [Op.iLike]: `%:name%` } },
```

then pass the `name` with sequelize replacement